### PR TITLE
planner: unify the parameters name in plan cache 

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Currently, we have many ways to represent the user defined variables such as `Args`, `Vars`, `UserVars`, `Param` and so on. We decide to uniformly name them as `..Param..` 
Issue Number: ref #36598

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by dayicklp — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]

File: planner/core/common_plans.go [hunk 1]
```diff
@@ -188,12 +188,12 @@ type Prepare struct {
...
 	baseSchemaProducer
 
-	Name         string
-	TxtProtoVars []expression.Expression
-	ExecID       uint32
-	Stmt         ast.StmtNode
-	StmtType     string
-	Plan         Plan
+	Name           string
+	TxtProtoParams []expression.Expression
```
Comment (qw4990) [inline planner/core/common_plans.go:192]: The prefix `TxtProto` can be removed.

Comment (fzzf678) [reply planner/core/common_plans.go]: > The prefix `TxtProto` can be removed.

PTAL again

**Review by qw4990 — APPROVED** [review]

**Pull request merged**